atr/sbom/tool.py

```diff
@@ -0,0 +1,108 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import datetime
+from typing import TYPE_CHECKING, Any, Final
+
+from . import maven, models
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+try:
+    from atr import metadata
+
+    atr_version = metadata.version
+except ImportError:
+    metadata = None
+    atr_version = "cli"
+
+_KNOWN_TOOLS: Final[dict[str, tuple[str, str, Callable[[str], str | None]]]] = {
```

Comment:
Maybe better to use a Pydantic model, dataclass, or namedtuple etc. for the values here.